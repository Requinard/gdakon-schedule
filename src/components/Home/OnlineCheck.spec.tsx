import { describe, it, expect, vi } from "vitest";

import { render, screen } from "../../test/test-utils";

import { OnlineCheck } from "./OnlineCheck";
describe("<OnlineCheck />", () => {
    it("renders nothing when online but error message is skipped", () => {
        vi.spyOn(navigator, "onLine", "get").mockReturnValue(true);

        render(<OnlineCheck />);
        expect(screen.queryByRole("alert")).toBeNull();
    });
    it("renders a success alert when online but not skipping", () => {
        vi.spyOn(navigator, "onLine", "get").mockReturnValue(true);

        render(<OnlineCheck showIfOnline />);
        expect(screen.getByRole("alert")).toMatchSnapshot();
        expect(screen.getByTestId("SuccessOutlinedIcon")).toBeVisible();
    });
    it("renders a warning when offline", () => {
        vi.spyOn(navigator, "onLine", "get").mockReturnValue(false);

        render(<OnlineCheck />);
        expect(screen.getByRole("alert")).toMatchSnapshot();
        expect(screen.getByTestId("ReportProblemOutlinedIcon")).toBeVisible();
    });
    it("renders a warning when offline and showOnline is set", () => {
        vi.spyOn(navigator, "onLine", "get").mockReturnValue(false);

        render(<OnlineCheck showIfOnline />);
        expect(screen.getByRole("alert")).toMatchSnapshot();
        expect(screen.getByTestId("ReportProblemOutlinedIcon")).toBeVisible();
    });
});
