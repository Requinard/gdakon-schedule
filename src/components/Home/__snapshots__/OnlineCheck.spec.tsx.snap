// Vitest Snapshot v1

exports[`<OnlineCheck /> > renders a success alert when online but not skipping 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-wnkddu-MuiPaper-root-MuiAlert-root"
  role="alert"
>
  <div
    class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
      data-testid="SuccessOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
      />
    </svg>
  </div>
  <div
    class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
  >
    <div
      class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
    >
      OnlineCheck.online.title
    </div>
    OnlineCheck.online.description
  </div>
</div>
`;

exports[`<OnlineCheck /> > renders a warning when offline 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-to2f46-MuiPaper-root-MuiAlert-root"
  role="alert"
>
  <div
    class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
      data-testid="ReportProblemOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
      />
    </svg>
  </div>
  <div
    class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
  >
    <div
      class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
    >
      OnlineCheck.offline.title
    </div>
    OnlineCheck.offline.description
  </div>
</div>
`;

exports[`<OnlineCheck /> > renders a warning when offline and showOnline is set 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-to2f46-MuiPaper-root-MuiAlert-root"
  role="alert"
>
  <div
    class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
      data-testid="ReportProblemOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
      />
    </svg>
  </div>
  <div
    class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
  >
    <div
      class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
    >
      OnlineCheck.offline.title
    </div>
    OnlineCheck.offline.description
  </div>
</div>
`;

exports[`<OnlineCheck /> > renders success when online 1`] = `
<div
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-to2f46-MuiPaper-root-MuiAlert-root"
  role="alert"
>
  <div
    class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
      data-testid="ReportProblemOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
      />
    </svg>
  </div>
  <div
    class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
  >
    <div
      class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
    >
      OnlineCheck.offline.title
    </div>
    OnlineCheck.offline.description
  </div>
</div>
`;
